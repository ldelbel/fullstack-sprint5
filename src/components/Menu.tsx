import React, { useContext } from "react";
import CategoriesContext from "../contexts/CategoriesContext";
import { ICategoriesContext } from "../types/interfaces";

interface IMenuItem {
  label: string;
}

function MenuItem({ label }: IMenuItem) {
  return (
    <li className="menu__item">
      <a className="menu__link" href="#home">
        <span>{label}</span>
      </a>
    </li>
  );
}

function Menu() {
  const { categories } = useContext<ICategoriesContext>(CategoriesContext);

  return (
    <nav className="header__menu menu">
      <ul className="menu__list">
        {categories.all &&
          categories.all.map((m) => <MenuItem key={m.id} label={m.label} />)}
      </ul>
    </nav>
  );
}

export default Menu;
