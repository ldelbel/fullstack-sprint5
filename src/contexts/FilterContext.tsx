import React from "react";
import { IFilterContext } from "../types/interfaces";

const FilterContext = React.createContext<IFilterContext>({} as IFilterContext);

export default FilterContext;
