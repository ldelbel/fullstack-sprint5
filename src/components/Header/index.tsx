import React, { ReactElement, useContext } from "react";
import FilterContext from "../../contexts/FilterContext";
import { IFilterContext } from "../../types/interfaces";
import Menu from "../Menu";
import { StyledHeader } from "./styled";

function Header(): ReactElement<React.FC> {
  const { setFilter } = useContext<IFilterContext>(FilterContext);

  return (
    <StyledHeader>
      <div className="header__container">
        <div className="header__mobile">
          <div className="header__drawer menu header__drawer--active">
            <img className="menu__img" src="assets/menu.svg" alt="menu" />
            <p className="menu__title">menu</p>
          </div>
          <div className="header__drawer close">
            <img src="assets/close.svg" alt="fechar menu" />
            <p className="close__title">fechar</p>
          </div>
          <h1 className="header__logo">
            <img className="header__img" src="assets/rchlo.svg" alt="Logo" />
          </h1>
        </div>
        <div className="header__desktop">
          <h1 className="header__logo">
            <img
              className="header__img"
              src="assets/riachuelo.svg"
              alt="Logo"
            />
          </h1>
        </div>
        <div className="header__search">
          <img className="header__icon" src="assets/search.svg" alt="lupa" />
          <input
            className="header__input"
            type="search"
            placeholder="O que você está procurando?"
            onChange={(event) => setFilter(event.target.value)}
          />
        </div>

        <Menu />
      </div>
    </StyledHeader>
  );
}

export default Header;
