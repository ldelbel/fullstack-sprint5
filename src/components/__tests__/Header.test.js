import React from "react";
import { render } from "@testing-library/react";
import Header from "../Header";
import FilterContext from "../../contexts/FilterContext";
import Menu from "../Menu";

jest.mock('../Menu')

const mockSetFilter = () => {};

describe("Header component", () => {
  it("should render categories", () => {

    Menu.mockImplementation( () => <div>Menu</div>)

    const { getByText } = render(
      <FilterContext.Provider value={{ mockSetFilter }}>
        <Header />
      </FilterContext.Provider>
    );
    expect(getByText("Menu")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    Menu.mockImplementation( () => <div>Menu</div>)
    const header = render(
      <FilterContext.Provider value={{ mockSetFilter }}>
        <Header />
      </FilterContext.Provider>
    );
    expect(header).toMatchSnapshot();
  });
});
