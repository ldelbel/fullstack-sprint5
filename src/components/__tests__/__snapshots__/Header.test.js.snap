// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdnxRM llzvJy"
      >
        <div
          class="header__container"
        >
          <div
            class="header__mobile"
          >
            <div
              class="header__drawer menu header__drawer--active"
            >
              <img
                alt="menu"
                class="menu__img"
                src="assets/menu.svg"
              />
              <p
                class="menu__title"
              >
                menu
              </p>
            </div>
            <div
              class="header__drawer close"
            >
              <img
                alt="fechar menu"
                src="assets/close.svg"
              />
              <p
                class="close__title"
              >
                fechar
              </p>
            </div>
            <h1
              class="header__logo"
            >
              <img
                alt="Logo"
                class="header__img"
                src="assets/rchlo.svg"
              />
            </h1>
          </div>
          <div
            class="header__desktop"
          >
            <h1
              class="header__logo"
            >
              <img
                alt="Logo"
                class="header__img"
                src="assets/riachuelo.svg"
              />
            </h1>
          </div>
          <div
            class="header__search"
          >
            <img
              alt="lupa"
              class="header__icon"
              src="assets/search.svg"
            />
            <input
              class="header__input"
              placeholder="O que você está procurando?"
              type="search"
            />
          </div>
          <div>
            Menu
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdnxRM llzvJy"
    >
      <div
        class="header__container"
      >
        <div
          class="header__mobile"
        >
          <div
            class="header__drawer menu header__drawer--active"
          >
            <img
              alt="menu"
              class="menu__img"
              src="assets/menu.svg"
            />
            <p
              class="menu__title"
            >
              menu
            </p>
          </div>
          <div
            class="header__drawer close"
          >
            <img
              alt="fechar menu"
              src="assets/close.svg"
            />
            <p
              class="close__title"
            >
              fechar
            </p>
          </div>
          <h1
            class="header__logo"
          >
            <img
              alt="Logo"
              class="header__img"
              src="assets/rchlo.svg"
            />
          </h1>
        </div>
        <div
          class="header__desktop"
        >
          <h1
            class="header__logo"
          >
            <img
              alt="Logo"
              class="header__img"
              src="assets/riachuelo.svg"
            />
          </h1>
        </div>
        <div
          class="header__search"
        >
          <img
            alt="lupa"
            class="header__icon"
            src="assets/search.svg"
          />
          <input
            class="header__input"
            placeholder="O que você está procurando?"
            type="search"
          />
        </div>
        <div>
          Menu
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
